src/ByteSizeLib/ByteSize.cs

```diff
@@ -305,9 +305,43 @@ public ByteSize Subtract(ByteSize bs)
 
         public static ByteSize Parse(string s)
         {
+            return TryParse(s).Value;
+        }
+
+        public static bool TryParse(string s, out ByteSize result)
+        {
+            var parseResult = TryParse(s, CachedError);
+            result = parseResult.GetValueOrDefault();
+            return !parseResult.IsError;
+        }
+
+        static readonly Exception CachedError = new Exception();
```

Comment:
Let me know if 4424a0bf16753d10e19cb5e8f4ff71f30a4984d7 feels less tricky. The difference is that `ParseResult` can now signal an error without an `Exception` by turning [`IsError` into a real state](https://github.com/atifaziz/ByteSize/blob/4424a0bf16753d10e19cb5e8f4ff71f30a4984d7/src/ByteSizeLib/ByteSize.cs#L329) as opposed to being computed off the null state of the exception. Previously, [a dummy exception](https://github.com/atifaziz/ByteSize/blob/bab4ab6ebacb1c68d2178a32281da6df168ebbc7/src/ByteSizeLib/ByteSize.cs#L318) was being used to the same effect. [`ParseResult` is now one Boolean heavier](https://github.com/atifaziz/ByteSize/blob/4424a0bf16753d10e19cb5e8f4ff71f30a4984d7/src/ByteSizeLib/ByteSize.cs#L333), and since it's a `struct`, needs to be passed to the [internal `TryParse`](https://github.com/atifaziz/ByteSize/blob/4424a0bf16753d10e19cb5e8f4ff71f30a4984d7/src/ByteSizeLib/ByteSize.cs#L352) as `Nullable<ParseResult>`, making it another Boolean heavier.

If you were okay with bab4ab6ebacb1c68d2178a32281da6df168ebbc7 in the first place, I can revert the 4424a0bf16753d10e19cb5e8f4ff71f30a4984d7 revision and it can be left archived in the repo if it ever needs to be restored. Your call.